libs/dh/wholesale/feature-calculations/src/lib/details/details.component.ts

```diff
@@ -81,21 +70,23 @@ import { WattModalActionsComponent, WattModalComponent } from '@energinet-datahu
 export class DhCalculationsDetailsComponent {
   private toast = inject(WattToastService);
   private transloco = inject(TranslocoService);
-  private apollo = inject(Apollo);
 
-  @Output() closed = new EventEmitter<void>();
   id = input<string>();
+  closed = output();
 
-  @ViewChild(WattDrawerComponent)
-  drawer!: WattDrawerComponent;
-
-  modal = viewChild.required<WattModalComponent>('modal');
+  drawer = viewChild(WattDrawerComponent);
+  modal = viewChild.required(WattModalComponent);
 
-  calculation?: Calculation;
-  error = false;
-  loading = false;
+  query = lazyQuery(GetCalculationByIdDocument, {
+    nextFetchPolicy: 'cache-first',
+  });
 
-  CalculationOrchestrationState = CalculationOrchestrationState;
+  result = computed(() => this.query.data()?.calculationById);
+  type = computed(() => this.result()?.calculationType ?? 'UNKNOWN');
+  executionType = computed(() => this.result()?.executionType);
+  statusType = computed(() => this.result()?.statusType ?? 'skeleton');
+  state = computed(() => this.result()?.state ?? 'INDETERMINATE');
+  isScheduled = computed(() => this.result()?.state === State.Scheduled);
```

Comment:
praise: nice change. Also quite cool that the properties will re-run/update only if the `result` Signal changes. No more worrying about excessive template re-rendering 🚀